ui: Handle Email signup alert message close

koorosh opened the pull request:
Resolves: #45457

- Auto close alert after 6 seconds;
- Close alert when Email Subscription form is unmounted;
- Close alert whet Email Subscription form is closed explicitly;

Release justification: bug fixes and low-risk updates to new functionality

koorosh committed:
ui: Handle Email signup alert message close

Release justification: bug fixes and low-risk updates to new functionality

- Auto close alert after 6 seconds;
- Close alert when Email Subscription form is unmounted;
- Close alert whet Email Subscription form is closed explicitly;

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/45674)
<!-- Reviewable:end -->

koorosh committed:
ui: Remove description and close button for Email signup alert messages

Release justification: bug fixes and low-risk updates to new functionality

- Show only title for Email signup alert message and remove close button;
- Align elements in Alert message according to designs;

Release note: None

dhartunian reviewed (approved):
> Reviewed 3 of 3 files at r1, 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/45674)* status: :shipit: complete! 0 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

dhartunian commented:
@koorosh I'm not sure this commit closes #45624. I think that ticket concerns the permanent closing of the call-to-action itself. I will comment in there regarding that issue.

dhartunian commented:
bors r+
